src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPCodeActionFixtureTestCase.java

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.fixtures;
+
+import com.google.gson.reflect.TypeToken;
+import com.intellij.codeInsight.daemon.impl.HighlightInfo;
+import com.intellij.codeInsight.intention.IntentionAction;
+import com.redhat.devtools.lsp4ij.JSONUtils;
+import com.redhat.devtools.lsp4ij.mock.MockLanguageServer;
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Base class test case to test LSP 'textDocument/codeAction' feature.
+ */
+public abstract class LSPCodeActionFixtureTestCase extends LSPCodeInsightFixtureTestCase {
+
+    public LSPCodeActionFixtureTestCase(String... fileNamePatterns) {
+        super(fileNamePatterns);
+    }
+
+    /**
+     * Test LSP code actions.
+     *
+     * @param fileName          the file name used to match registered language servers.
+     * @param editorContentText the editor content text.
+     * @param codeActionsJson   the LSP CodeAction list as a JSON string.
+     * @param expectedActions   the expected IntelliJ intention action text.
+     */
+    protected void assertCodeActions(@NotNull String fileName,
```

Comment:
Change signature like this:

`protected List<CodeAction> assertCodeActions(...`